python/llama-index-server/llama_index/server/api/routers/dev.py

```
@@ -0,0 +1,104 @@
+            tmp.write(update.content)
+            tmp_path = tmp.name
+        try:
+            # Validate workflow file using the actual callable name from the workflow_factory
+            factory_func_name = server_settings.workflow_factory_signature
+            print(f"Validating workflow file: {tmp_path}")
+            print(f"Factory function name: {factory_func_name}")
+            validate_workflow_file(
+                workflow_path=tmp_path, factory_signature=factory_func_name
+            )
+        except Exception as e:
+            raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)